### PR TITLE
feat: add internal/external Postgres config

### PR DESCRIPTION
**Pull request by pmoscode**
## Description

Adds the ability to configure the PostgreSQL DB. Now, you can bring your own PostgreSQL or use the bundled one.
And you can also predefine the DB password or let PostgreSQL create one for you.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up to date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] Copyright and license header are present on all affected files

### CODE REVIEW

**Comment by pmoscode** [comment]
@borisrizov-zf @mknoopvw 
Please, review the changes.

**Comment by pmoscode** [comment]
@SebastianBezold 
Please merge this PR after review is done.
Thx

File: charts/managed-identity-wallet/tests/default/deployment_test.yaml [hunk 1]
```diff
@@ -99,7 +99,7 @@ tests:
           path: spec.template.spec.containers[0].env
       - lengthEqual:
           path: spec.template.spec.containers[0].env
-          count: 1
+          count: 9
```
Comment (borisrizov-zf) [inline charts/managed-identity-wallet/tests/default/deployment_test.yaml:102]: We shouldn't use magic numbers in tests, add a comment describing the origin of `9`

File: charts/managed-identity-wallet/values.yaml [hunk 2]
```diff
@@ -23,40 +23,53 @@ replicaCount: 1
...
 nameOverride: ""
 fullnameOverride: ""
 
-# -- Parameters for the application (will be stored as secrets - so, for passwords, ...)
-secrets: {}
-#  DB_USER_NAME: xxx # Has to be the same as below in "postgresql" section
-#  DB_PASSWORD: xxx # Has to be the same as below in "postgresql" section
-#  KEYCLOAK_CLIENT_ID: xxx
-#  ENCRYPTION_KEY: xxx  # 64 chars
+# -- Parameters for the application (will be stored as secrets - so, for important values, ...)
```
Comment (borisrizov-zf) [inline charts/managed-identity-wallet/values.yaml:38]: The sentence is confusing. Please reword.

**Review by borisrizov-zf — APPROVED** [review]

**Comment by SebastianBezold** [comment]
HI all,
there are again valuable comments to the change from @borisrizov-zf, but the PR is approved anyways. 
Just as suggestion, I would rather address or fix the comments right in this PR (preferred), or at least create issues as a reminder of the spots, that can be improved.
I think if i'll merge the PR without addressing the comments or creating issues, the valuable feedback will be lost

File: .github/workflows/chart-lint.yml [hunk 3]
```diff
@@ -1,68 +1,87 @@
...
+     - name: Install Helm unittest plugin
+       run: |
+         helm plugin install https://github.com/helm-unittest/helm-unittest.git
 
-#       - name: Set up chart-testing
-#         uses: helm/chart-testing-action@v2.3.1
+     # Setup python as a prerequisite for chart linting
+     - uses: actions/setup-python@v2
+       with:
+         python-version: 3.7
```
[the remark this answers is not in the document]
Comment (pmoscode) [reply .github/workflows/chart-lint.yml]: Done.

File: charts/managed-identity-wallet/tests/default/deployment_test.yaml (hunk 1 above)
Comment (pmoscode) [reply charts/managed-identity-wallet/tests/default/deployment_test.yaml]: Done. And fixed a small bug. :)

File: charts/managed-identity-wallet/values.yaml (hunk 2 above)
Comment (pmoscode) [reply charts/managed-identity-wallet/values.yaml]: Done.

File: charts/managed-identity-wallet/values.yaml [hunk 4]
```diff
@@ -23,40 +23,53 @@ replicaCount: 1
...
+secrets:
+  # -- Keycloak confidential client id for MIW application
+  KEYCLOAK_CLIENT_ID: ""
+  # -- AES encryption key used to encrypt/decrypt private keys (random 32 chars)
```
[the remark this answers is not in the document]
Comment (pmoscode) [reply charts/managed-identity-wallet/values.yaml]: Done.

**Comment by mknoopvw** [comment]
I am just curious:
In the "{stage}-values.yaml" were "" added for some values: is this to fix the fault or just to unify the appearance?
Thank you.

File: charts/managed-identity-wallet/values-dev.yaml [hunk 5]
```diff
@@ -21,16 +21,13 @@ image:
   tag: "latest-java-did-web"
 
 secrets:
-  DB_USER_NAME: <path:managed-identity-wallets/data/dev/postgres-managed-identity-wallets-secret-config#user>
-  DB_PASSWORD: <path:managed-identity-wallets/data/dev/postgres-managed-identity-wallets-secret-config#password>
-  ENCRYPTION_KEY: 76a7834fb37e090c2789a9b1a76748d3
+  ENCRYPTION_KEY: "76a7834fb37e090c2789a9b1a76748d3"
```
[the remark this answers is not in the document]
Comment (borisrizov-zf) [reply charts/managed-identity-wallet/values-dev.yaml:24]: According to yaml, no. But it's a good idea when special characters might appear in the value.

**Review by borisrizov-zf — APPROVED** [review]

**Review by mknoopvw — APPROVED** [review]

**Comment by pmoscode** [comment]
> HI all,
> there are again valuable comments to the change from @borisrizov-zf, but the PR is approved anyways.
> Just as suggestion, I would rather address or fix the comments right in this PR (preferred), or at least create issues as a reminder of the spots, that can be improved.
> I think if i'll merge the PR without addressing the comments or creating issues, the valuable feedback will be lost

Hi @SebastianBezold,
sorry for the misleading post. First all reviews should be green, and then...
Now, it's done. 
Fyi: The "Lint and Test Chart" check is failing, because the image isn't available on hub.docker.com yet.

Thx

**Comment by pmoscode** [comment]
> > > HI all,
> > > there are again valuable comments to the change from @borisrizov-zf, but the PR is approved anyways.
> > > Just as suggestion, I would rather address or fix the comments right in this PR (preferred), or at least create issues as a reminder of the spots, that can be improved.
> > > I think if i'll merge the PR without addressing the comments or creating issues, the valuable feedback will be lost
> > 
> > 
> > Hi @SebastianBezold, sorry for the misleading post. First all reviews should be green, and then... Now, it's done. Fyi: The "Lint and Test Chart" check is failing, because the image isn't available on hub.docker.com yet.
> > Thx
> 
> Hi @pmoscode! The "Lint and Test Charts" workflow fails because one of the container environment variables in the `values.yaml` is invalid. See the [error description](https://github.com/eclipse-tractusx/managed-identity-wallet/actions/runs/5794708643/job/15704829066?pr=48#step:13:33).

Yes, you're right. Fixed it. But the test is still failing, because the corresponding image isn't released on Docker Hub yet. Pipeline is still in dev.

**Comment by pmoscode** [comment]
@almadigabor 
Fix is done.

**Comment by almadigabor** [comment]
Merged this PR but raised a new issue #51, so the helm test workflow can run properly even when the new image version is not yet released.

**Pull request merged**